sdk/translation/Azure.AI.Translation.Document/src/DocumentTranslationInput.cs

```diff
@@ -50,13 +50,16 @@ public DocumentTranslationInput(Uri sourceUri, Uri targetUri, string targetLangu
         /// <a href="https://docs.microsoft.com/azure/cognitive-services/translator/language-support#translate"/>.</param>
         /// <param name="glossary">Custom <see cref="TranslationGlossary"/> to be used in the translation operation. For supported file types see
         /// <see cref="DocumentTranslationClient.GetSupportedGlossaryFormatsAsync(System.Threading.CancellationToken)"/>.</param>
-        public void AddTarget(Uri targetUri, string languageCode, TranslationGlossary glossary = default)
+        /// <param name="categoryId">Category/custom model ID for using custom translation.</param>
+        public void AddTarget(Uri targetUri, string languageCode, TranslationGlossary glossary = default, string categoryId = default)
```

Comment:
From a technical standpoint, this looks fine.   I'm not sure how I feel about the double optional.    Should we consider possibly making an overload with `categoryId` so that it's more discoverable, or maybe removing the optionals and adding an overload that takes a fully-formed `TranslationTarget`?

Reply:
The idea behind this method is to use the progressive disclosure principle where a user already used the service to translate a document from one language to another and now wants to add another language to the translation.
Originally, we had the 2 required parameters plus a less common one which are the glossaries in case the user wanted to explore that option too. This made it look like there was no other way to also instantiate the `categoryId` using this flow.

> making an overload with categoryId so that it's more discoverable

Is this something we are applying in our libraries thanks to the studies we've done? I've been in some studies where optional parameters helped more than overloads but this was before my leave so I might be missing context here.

Reply:
It's a pattern that we've been applying to T2 when there are core scenarios where less experienced developers are included in the audience.   It's not explicitly called out in the guidelines, but tends to come up during archboard reviews.   The best example is the constructor patterns that we use.    If we feel that this is an advanced scenario, then I don't think we'll see concern with multiple optional.

Reply:
Interesting. Thank you so much for sharing that.

This is def for advanced users. CategoryId means the user previously created a custom translation and now wants to use that model.